src/aiAgentsConfiguration/aiAgentHooks.ts

```diff
@@ -0,0 +1,299 @@
+/* --------------------------------------------------------------------------------------------
+ * SonarLint for VisualStudio Code
+ * Copyright (C) 2017-2025 SonarSource SA
+ * sonarlint@sonarsource.com
+ * Licensed under the LGPLv3 License. See LICENSE.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as vscode from 'vscode';
+import * as path from 'node:path';
+import * as fs from 'node:fs';
+import { SonarLintExtendedLanguageClient } from '../lsp/client';
+import { Commands } from '../util/commands';
+import { AGENT } from './aiAgentUtils';
+
+interface HookConfig {
+  command: string;
+  show_output?: boolean;
+  working_directory?: string;
+}
+
+interface HooksJson {
+  hooks: {
+    pre_read_code?: HookConfig[];
+    post_write_code?: HookConfig[];
+  };
+}
+
+const HOOK_SCRIPT_PERMISSIONS = 0o700;
+const HOOK_SCRIPT_PATTERN = /sonarqube_analysis_hook\.(js|py|sh)$/;
+
+export function getCurrentAgentWithHookSupport(): AGENT | undefined {
+  const appName = vscode.env.appName.toLowerCase();
+  // Hooks are only available on windsurf-next (beta) for now
+  if (appName.includes('windsurf') && appName.includes('next')) {
+    return AGENT.WINDSURF;
+  }
+  return undefined;
+}
+
+function getWindsurfDirectory(): string {
+  const appName = vscode.env.appName.toLowerCase();
+  if (appName.includes('next')) {
+    return 'windsurf-next';
+  }
+  return 'windsurf';
+}
+
+function getHooksConfigPath(agent: AGENT): string | undefined {
+  switch (agent) {
+    case AGENT.WINDSURF: {
+      const homeDir = process.env.HOME || process.env.USERPROFILE;
+      if (!homeDir) {
+        return undefined;
+      }
+      const windsurfDir = getWindsurfDirectory();
+      return path.join(homeDir, '.codeium', windsurfDir, 'hooks.json');
+    }
+    case AGENT.CURSOR:
+      return undefined;
+    default:
+      return undefined;
+  }
+}
+
+function getHookScriptDirectory(agent: AGENT): string | undefined {
+  switch (agent) {
+    case AGENT.WINDSURF: {
+      const homeDir = process.env.HOME || process.env.USERPROFILE;
+      if (!homeDir) {
+        return undefined;
+      }
+      const windsurfDir = getWindsurfDirectory();
+      return path.join(homeDir, '.codeium', windsurfDir, 'hooks');
+    }
+    case AGENT.CURSOR:
+      return undefined;
+    default:
+      return undefined;
+  }
+}
+
+async function readHooksConfig(configPath: string): Promise<HooksJson> {
+  try {
+    const content = await fs.promises.readFile(configPath, 'utf8');
+    return JSON.parse(content);
+  } catch {
+    return { hooks: {} };
+  }
+}
+
+async function writeHooksConfig(configPath: string, config: HooksJson): Promise<void> {
+  const dir = path.dirname(configPath);
+  await fs.promises.mkdir(dir, { recursive: true });
+  await fs.promises.writeFile(configPath, JSON.stringify(config, null, 2), 'utf8');
```

Comment:
What if file already exists & there are some hooks defined inside? Will we rewrite everything?